packages/gatsby/src/commands/develop.ts

```diff
@@ -163,6 +164,17 @@ const REGEX_IP = /^(?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.){3}(?:25
 
 module.exports = async (program: IProgram): Promise<void> => {
   const developProcessPath = slash(require.resolve(`./develop-process`))
+
+  try {
+    program.port = await detectPortInUseAndPrompt(program.port)
+  } catch (e) {
+    if (e.message === `USER_REJECTED`) {
+      process.exit(0)
+    }
+
+    throw e
+  }
+
```

Comment:
👍 looks like the same code from [before](https://github.com/gatsbyjs/gatsby/blob/4fd8f8e463402bff2944d98d83f2920dc0792755/packages/gatsby/src/commands/develop.ts#L390-L398)

Reply:
Yes, it is. This PR is restoring that, as pointed out seems like it was accidentally removed.